### PR TITLE
feat: add contract for ERC20KavaWrappedNativeCoin

### PR DESCRIPTION
**Pull request by pirtleshell**
## Description
Configures a hardhat project for developing and testing solidity contracts in this repo.
Includes ERC20KavaWrappedNativeCoin contract to be used by evmutil to wrap sdk.Coins.

## Checklist
 - [x] ~~Changelog has been updated as necessary.~~

### CODE REVIEW

**Review by galxy25 — APPROVED** [review]

File: contracts/contracts/ERC20KavaWrappedNativeCoin.sol [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
```
Comment (pirtleshell) [inline contracts/contracts/ERC20KavaWrappedNativeCoin.sol:2]: this is the contract that review should be focused on 😄

File: contracts/package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  "name": "kava-contracts",
+  "version": "0.0.1",
+  "author": "Kava Labs",
+  "private": true,
+  "description": "Solidity contracts for Kava Blockchain",
+  "engines": {
+    "node": ">=18.0.0"
+  },
+  "scripts": {
+    "build": "npm run clean && npm run compile && npm run ethermint-json",
```
Comment (pirtleshell) [inline contracts/package.json:11]: as mentioned in the readme, this is the command that will compile the contract and convert it to the necessary json format for use in evmutil.

File: x/evmutil/types/ethermint_json/ERC20KavaWrappedNativeCoin.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (pirtleshell) [inline x/evmutil/types/ethermint_json/ERC20KavaWrappedNativeCoin.json:1]: this file is created by the `npm run build` command inside the hardhat project.

File: contracts/hardhat.config.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    settings: {
+      // istanbul upgrade occurred before the london hardfork, so is compatible with kava's evm
+      evmVersion: "istanbul",
+      // optimize build for deployment to mainnet!
+      optimizer: {
+        enabled: true,
+        runs: 1000,
+      },
+    },
+  },
```
Comment (pirtleshell) [inline contracts/hardhat.config.ts:16]: here are the solc configurations and optimizations used to compile the contract.

File: contracts/package.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    "lint-fix": "eslint '**/*.{js,ts}' --fix",
+    "prettier": "prettier '**/*.{json,sol,md}' --check",
+    "prettier-fix": "prettier '**/*.{json,sol,md}' --write",
+    "solhint": "solhint 'contracts/**/*.sol' --max-warnings 0",
+    "solhint-fix": "solhint 'contracts/**/*.sol' --fix",
+    "test": "hardhat test"
+  },
+  "devDependencies": {
+    "@nomicfoundation/hardhat-toolbox": "^2.0.2",
+    "@openzeppelin/contracts": "4.8.3",
```
Comment (pirtleshell) [inline contracts/package.json:27]: using a fixed verison of the open zeppelin contracts package (note the lack of `^` 🥕 )

File: contracts/scripts/deploy.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { ethers } from "hardhat";
+
+async function main() {
```
Comment (drklee3) [inline contracts/scripts/deploy.ts:3]: We could also maybe move the deploy/mint tasks over from bridge to here too, so the CI seed script can just use this repo. Useful to have a way to easily deploy contracts with any name/symbol/decimals

https://github.com/Kava-Labs/kava-bridge/blob/main/contract/tasks/deployERC20.ts

File: contracts/test/ERC20KavaWrappedNativeCoin.test.ts [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+      await expect(tx).to.not.be.reverted;
+
+      const bal = await erc20.balanceOf(await sender.getAddress());
+      expect(bal).to.equal(amount);
+    });
+  });
+
+  describe("burn", function () {
+    it("should reject non-owner", async function () {
+      const tx = erc20.connect(sender).mint(await sender.getAddress(), 10n);
```
Comment (drklee3) [inline contracts/test/ERC20KavaWrappedNativeCoin.test.ts:50]:
```suggestion
      const tx = erc20.connect(sender).burn(await sender.getAddress(), 10n);
```

**Review by drklee3 — APPROVED** [review]
looks good, just a minor note with test

File: contracts/test/ERC20KavaWrappedNativeCoin.test.ts (hunk 7 above)
Comment (pirtleshell) [reply contracts/test/ERC20KavaWrappedNativeCoin.test.ts:50]: nice catch, thanks!

File: contracts/scripts/deploy.ts (hunk 6 above)
Comment (pirtleshell) [reply contracts/scripts/deploy.ts:3]: good idea, it would be nice to remove kava-bridge from this repo's CI. i created a ticket for this work: [sc9364]

**Pull request merged**